core/src/main/java/com/linecorp/armeria/server/ExceptionReportingServerErrorHandler.java

```diff
@@ -0,0 +1,113 @@
+ * When services are not annotated with {@link LoggingService}, exceptions are not caught be default.
+ * {@link ExceptionReportingServerErrorHandler} is used to catch those uncaught exceptions.
+ */
+class ExceptionReportingServerErrorHandler implements ServerErrorHandler {
+    private static final Logger logger = LoggerFactory.getLogger(ExceptionReportingServerErrorHandler.class);
+    private final ServerErrorHandler delegate;
+    private final LongAdder counter;
+    private final long intervalInSeconds;
+    private boolean isScheduled;
+    @Nullable
+    private Throwable lastThrownException;
+    @Nullable
+    private ScheduledFuture<?> reportExceptionsSchedule;
+
+    ExceptionReportingServerErrorHandler(ServerErrorHandler serverErrorHandler, long intervalInSeconds) {
+        delegate = serverErrorHandler;
+        this.intervalInSeconds = intervalInSeconds;
+        counter = new LongAdder();
+    }
+
+    /**
+     * If it's required to log exceptions, increase the counter and store last thrown exception.
+     */
+    @Nullable
+    @Override
+    public HttpResponse onServiceException(ServiceRequestContext ctx, Throwable cause) {
+        if (isScheduled && ctx.shouldLogException()) {
+            lastThrownException = cause;
+            counter.increment();
+        }
+        return delegate.onServiceException(ctx, cause);
+    }
+
+    @Nullable
+    @Override
+    public AggregatedHttpResponse renderStatus(ServiceConfig config, @Nullable RequestHeaders headers,
+                                               HttpStatus status, @Nullable String description,
+                                               @Nullable Throwable cause) {
+        return delegate.renderStatus(config, headers, status, description, cause);
+    }
+
+    /**
+     * Starts logging exceptions.
+     * @param workerGroup eventLoopGroup on which logging exceptions will be scheduled.
+     */
+    void start(EventLoopGroup workerGroup) {
+        requireNonNull(workerGroup, "workerGroup");
+        if (!isScheduled && intervalInSeconds > 0) {
+            reportExceptionsSchedule = workerGroup.scheduleAtFixedRate(
+                    this::logExceptions, intervalInSeconds, intervalInSeconds, TimeUnit.SECONDS);
+            isScheduled = true;
+        }
+    }
+
+    /**
+     * Stops logging exceptions.
+     */
+    void stop() {
+        if (isScheduled && reportExceptionsSchedule != null) {
+            reportExceptionsSchedule.cancel(true);
+            isScheduled = false;
+        }
+    }
+
+    private void logExceptions() {
+        final long totalExceptions = counter.sumThenReset();
+        if (totalExceptions != 0) {
+            logger.warn("Observed {} uncaught exceptions in last {} seconds. " +
+                        "If you don't see error logs please use LoggingService decorator. " +
+                        "Last thrown exception is:",
+                        totalExceptions, intervalInSeconds, lastThrownException);
```

Comment:
I guess the chance is very low, but we can be more defensive:

```java
final Throwable lastThrownException = this.lastThrownException;
if (lastThrownException != null) {
   ... log with last thrown exception ...
   this.lastThrownException = null;
} else {
   ... log without last thrown exception ...
}
```

Reply:
I've changed the logging behavior to log `lastThrownException` only when it's not null.